akka-actor-tests/src/test/scala/akka/io/dns/internal/AsyncDnsResolverSpec.scala

```diff
@@ -197,22 +198,22 @@ class AsyncDnsResolverSpec extends AkkaSpec("""
       val recordTtl = Ttl.fromPositive(100.seconds)
 
       val ipv4Record = ARecord("cats.com", recordTtl, InetAddress.getByName("127.0.0.1"))
-      dnsClient1.expectMsg(Question4(1, "cats.com"))
-      dnsClient1.reply(Answer(1, im.Seq(ipv4Record)))
 
       r ! Resolve("cats.com", Ip(ipv4 = true, ipv6 = false))
+      dnsClient1.expectMsg(Question4(1, "cats.com"))
+      dnsClient1.reply(Answer(1, im.Seq(ipv4Record)))
       senderProbe.expectMsg(Resolved("cats.com", im.Seq(ipv4Record)))
 
+      r ! Resolve("cats.com", Ip(ipv4 = true, ipv6 = false))
       dnsClient1.expectNoMessage(50.millis)
 
-      r ! Resolve("cats.com", Ip(ipv4 = true, ipv6 = false))
       senderProbe.expectMsg(Resolved("cats.com", im.Seq(ipv4Record)))
 
+      Thread.sleep(5000)
```

Comment:
Missed this the last review, this slow a test case is not good. Tweak the caching down to a much lower value, max 1s, but preferrably even lower than that.

Reply:
Good catch, after the above change ^ then https://github.com/akka/akka/pull/27598#issuecomment-530437046.